test/parse.js

```diff
@@ -429,6 +429,14 @@ test('parse()', function (t) {
         st.end();
     });
 
+    t.test('parses comma delimited array while having percent-encoded comma treated as normal text', function (st) {
+        st.deepEqual(qs.parse('foo=a%2Cb', { comma: true }), { foo: ['a', 'b'] });
```

Comment:
@ljharb this test is wrong, it should return "a,b". It's the main point of the pull request and the related issue.

Reply:
ah, i understand. my bad.

Reply:
Filed https://github.com/ljharb/qs/pull/361 to proceed with a proper fix.